server/svix-server/src/main.rs

```diff
@@ -90,19 +94,58 @@ async fn main() {
         );
     }
 
+    let otel_layer = cfg.opentelemetry_address.as_ref().map(|addr| {
+        // Configure the OpenTelemetry tracing layer
+        opentelemetry::global::set_text_map_propagator(
+            opentelemetry::sdk::propagation::TraceContextPropagator::new(),
+        );
+
+        let exporter = opentelemetry_otlp::new_exporter()
+            .tonic()
+            .with_endpoint(addr);
+
+        let tracer = opentelemetry_otlp::new_pipeline()
+            .tracing()
+            .with_exporter(exporter)
+            .with_trace_config(
+                opentelemetry::sdk::trace::config()
+                    .with_sampler(
+                        cfg.opentelemetry_sample_ratio
+                            .map(opentelemetry::sdk::trace::Sampler::TraceIdRatioBased)
+                            .unwrap_or(opentelemetry::sdk::trace::Sampler::AlwaysOn),
+                    )
+                    .with_resource(opentelemetry::sdk::Resource::new(vec![
+                        opentelemetry::KeyValue::new("service.name", "svix_server"),
+                    ])),
+            )
+            .install_batch(Tokio)
+            .unwrap();
+        tracing_opentelemetry::layer().with_tracer(tracer)
+    });
+
+    // Then initialize logging with an additional layer priting to stdout. This additional layer is
+    // either formatted normally or in JSON format
     match cfg.log_format {
         cfg::LogFormat::Default => {
-            tracing_subscriber::fmt::init();
+            let stdout_layer = tracing_subscriber::fmt::layer();
+            tracing_subscriber::Registry::default()
+                .with(otel_layer)
+                .with(stdout_layer)
+                .with(tracing_subscriber::EnvFilter::from_default_env())
+                .init();
```

Comment:
This looks like exactly the same code as below, shouldn't this be shared between the two log formats?
I'm thinking:
```rust
// Is it stdout or stderr?
let stdout_layer = match cfg.log_format {
  ...
};
// Probably also dependent on whether the otel_layer exists
tracing_subscriber::Registry::default()
                .with(otel_layer)
                .with(stdout_layer)
                .with(tracing_subscriber::EnvFilter::from_default_env())
                .init();
```

Reply:
The `tracing_subscriber`layers are different types depending on log format, so the initialization has to occur there.

Reply:
It looks identical to me. The only difference I see is stdout_layer, which my comment above addresses.